File: play-interop-test-java/src/main/java/controllers/GreeterServiceImpl.java
/*
 * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
 */

// #service-impl
package controllers; // TODO: Move into 'services' package?

import example.myapp.helloworld.grpc.GreeterService;
import example.myapp.helloworld.grpc.HelloReply;
import example.myapp.helloworld.grpc.HelloRequest;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

/** User implementation, with support for dependency injection etc */
@Singleton
public class GreeterServiceImpl implements GreeterService {

  @Inject
  public GreeterServiceImpl() {
  }


  // FIXME: We don't actually need a Materializer here for this example, but should we include it anyway?
  //  @Inject
  //  public GreeterServiceImpl(Materializer mat) {
  //    super(mat);
  //  }

  @Override
  public CompletionStage<HelloReply> sayHello(HelloRequest in) {
    String message = String.format("Hello, %s!", in.getName());
    HelloReply reply = HelloReply.newBuilder().setMessage(message).build();
    return CompletableFuture.completedFuture(reply);
  }

}
// #service-impl

File: play-interop-test-scala/src/main/scala/controllers/GreeterServiceImpl.scala
/*
 * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
 */

// #service-impl
package controllers

import akka.stream.Materializer
import example.myapp.helloworld.grpc.helloworld.{ AbstractGreeterServiceRouter, HelloReply, HelloRequest }
import javax.inject.{ Inject, Singleton }

import scala.concurrent.Future

/** User implementation, with support for dependency injection etc */
@Singleton
class GreeterServiceImpl @Inject() (implicit mat: Materializer) extends AbstractGreeterServiceRouter(mat) {

  override def sayHello(in: HelloRequest): Future[HelloReply] = Future.successful(HelloReply(s"Hello, ${in.name}!"))

}
// #service-impl
